graphql/redact_secrets_plugin.go

```diff
@@ -0,0 +1,129 @@
+package graphql
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+)
+
+// GenerateSecretFields generates a file that contains a list of fields that should be redacted in logs.
+func GenerateSecretFields(cfg *config.Config) error {
+	redactedFields := make(map[string]bool)
+
+	for _, schemaType := range cfg.Schema.Types {
+		for _, field := range schemaType.Fields {
+			if field.Directives.ForName("redactSecrets") != nil {
+				redactedFields[field.Name] = true
+			}
+		}
+	}
+
+	var fields []string
+	for field := range redactedFields {
+		fields = append(fields, field)
+	}
+
+	// Sort the fields to ensure consistent output.
+	sort.Strings(fields)
+
+	return generateRedactedFieldsFile(fields)
+}
+
+// generateRedactedFieldsFile generates a file that contains a list of fields that should be redacted in logs. It is used to generate the redacted_fields_gen.go file.
+// The file contains a list of fields that should be redacted in logs, a function to check if a field should be redacted, and a function to redact fields in a map.
+func generateRedactedFieldsFile(fields []string) error {
+	file, err := os.Create("graphql/redacted_fields_gen.go")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(`// Code generated by graphql/redact_secrets_plugin.go DO NOT EDIT.
+package graphql
+
+var redactedFields = map[string]bool{
+`)
+	if err != nil {
+		return err
+	}
+
+	for _, field := range fields {
+		_, err = file.WriteString(fmt.Sprintf("\t\"%s\": true,\n", field))
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = file.WriteString("}\n\n")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func isFieldRedacted(fieldName string, fieldsToRedact map[string]bool) bool {
+	_, ok := fieldsToRedact[fieldName]
+	return ok
+}
```

Comment:
is there an advantage to checking the field is in the map rather than just checking the value in the map? (since values not in the map will return false as well)